src/core/model_imp.cpp

```diff
@@ -65,16 +65,13 @@ std::string ModelImp::decode_iter(int& token) {
 
 int32_t ModelImp::sample_and_update() {
     // sample the next token
-   auto token = llama_sample_top_p_top_k(
-           *m_vocab, m_logist.data(), m_last_queue, m_repeat_penalty, m_top_k, m_top_p,
-           m_temp, m_rng);
+    auto token = llama_sample_top_p_top_k(
+            *m_vocab, m_logist.data(), m_last_queue, m_repeat_penalty, m_top_k, m_top_p,
+            m_temp, m_rng);
     // update the last queue
     m_last_queue.push_back(token);
     m_last_queue.pop_front();
     m_pre_token = token;
-    if (token == m_end_token) {
```

Comment:
keep this, because some other apps, like alpaca don't call the model->deactive(), thus when no chat,  all the thread  are in busy wait status

Reply:
I added it in the application, check https://github.com/MegEngine/InferLLM/pull/62/files#diff-53d392bcfa35505c546c94f5d0eab505967ba1dcaa4a4ca47904794c5095c894R248.

So, in theory, when app is waiting for user input, it stops.

Reply:
yes，use condition_variable in every decode iteration is slow， here only when recieve the end_token it will deactive, it is the same with `when app is waiting for user input, it stops.`